sdk/java/dagger-java-annotation-processor/src/test/resources/io/dagger/gen/entrypoint/entrypoint.java

```diff
@@ -133,18 +135,24 @@ private ModuleID register() throws ExecutionException, DaggerQueryException,
                         .withArg("a", dag.typeDef().withKind(TypeDefKind.FLOAT_KIND))
                         .withArg("b", dag.typeDef().withKind(TypeDefKind.FLOAT_KIND)))
                 .withField("source", dag.typeDef().withObject("Directory"), new TypeDef.WithFieldArguments().withDescription("Project source directory"))
-                .withField("version", dag.typeDef().withKind(TypeDefKind.STRING_KIND)));
+                .withField("version", dag.typeDef().withKind(TypeDefKind.STRING_KIND))
+                .withConstructor(
+                    dag.function("New",
```

Comment:
I don't think it makes a difference in practice but can be confusing when other people try to understand from this. Constructor name should be empty:
```suggestion
                    dag.function("",
```

Reply:
ok, I'll change that, to have it right from start.

But I took it from the generated Go code. Is there a place where those rules are defined?

For instance if you `dagger develop` in `modules/go` you'll get:

```go
WithConstructor(
    dag.Function("New",
        dag.TypeDef().WithObject("Go")).
```

Reply:
> For instance if you `dagger develop` in `modules/go` you'll get:
> 
> ```go
> WithConstructor(
>     dag.Function("New",
>         dag.TypeDef().WithObject("Go")).
> ```

Oh, I wasn't actually aware of that. At one point a constructor was registered by a function with an empty name. Just like the way you know to call it is via a `FunctionCall.name` with empty value. 

Now with the `withConstructor` call it doesn't really matter, and is wiped server side:

https://github.com/dagger/dagger/blob/1d7fa24fb5d02ffefaf9760fa92df70c2ab28c6a/core/schema/module.go#L298-L301

I suppose in Go it was just easier to leave the function name in to reuse the general function to typedef conversion (that's why the name is "New", because in Go the constructor is always called "New"), but I still prefer not to register with a name so it doesn't confuse others that it's something that may be needed. Especially someone looking into adding modules support to an SDK and is looking to other SDKs for comparison 🙂